packages/application-shell/src/components/requests-in-flight-loader/reducer.ts

```diff
@@ -0,0 +1,41 @@
+import { Action } from 'redux';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import { SHOW_LOADING, HIDE_LOADING } from '@commercetools-frontend/constants';
+import {
+  TShowRequestInFlightAction,
+  THideRequestInFlightAction,
+} from './types';
+
+const isShowRequestInFlightAction = (
+  action: Action
+): action is TShowRequestInFlightAction => action.type === SHOW_LOADING;
+const isHideRequestInFlightAction = (
+  action: Action
+): action is THideRequestInFlightAction => action.type === HIDE_LOADING;
+
+const excludeFirstOccurrence = <Item>(list: Item[], item: Item) => {
+  const index = list.indexOf(item);
+  return [...list.slice(0, index), ...list.slice(index + 1)];
+};
+
+export default (requestsInFlight?: string[], action?: Action) => {
```

Comment:
Nit: should we type this to `type RequestInFlight = string` and `type RequestsInFlight = RequestInFlight[]`?